File: nvxapp.client/src/nvx/ClientServer-Service/Account/Models/company-model.ts
import { ModelResult } from "../../ModelsBase/model-result";
import { RolesModel } from "./user-roles-model";


export class CompanyListModel {
  constructor(
    public idCompany: number = 0,
    public descrizione: string | null = "",
    public idAspNetUsers: string = "",
    public mainUser: boolean = false
  ) { }
}
export class CompanyListInModel {
  
}
export class CompanyListOutModel extends ModelResult {

  public companyList: CompanyListModel[] ;

}



export class CompanyEditModel {
  constructor(
    public idCompany: number = 0,
    public descrizione: string | null = "",
    public idAspNetUsers: string = "",
    public mainUser: boolean = false,
    public mail: string | null = null,
    public pw: string | null = null
  ) { }
}
export class CompanyGetInModel {
  public id: number;
}
export class CompanyGetOutModel extends ModelResult {

  public companyEdit: CompanyEditModel;

}
export class CompanyPutInModel  {

  public companyEdit: CompanyEditModel;

}
export class CompanyPutOutModel extends ModelResult {

  public companyEdit: CompanyEditModel;

}

File: nvxapp.server.service/ClientServer-Service/Account/Models/CompanyModel.cs
using nvxapp.server.data.Entities.Public;
using nvxapp.server.service.ClientServer_Service.ModelsBase;
using System;
using System.Collections.Generic;
using System.ComponentModel.DataAnnotations;
using System.Linq;
using System.Text;
using System.Threading.Tasks;
using static nvxapp.server.data.Entities.AspNetUsersDataUtil;

namespace nvxapp.server.service.ClientServer_Service.Account.Models
{
    public class CompanyModel
    {
        public int IdCompany { get; set; }

        public required string IdAspNetUsers { get; set; }

        public string? Descrizione { get; set; }

    }
    public class CompanyListInModel
    {
    }
    public class CompanyListOutModel : ModelResult
    {
        public List<CompanyModel> CompanyList { get; set; }

        public CompanyListOutModel()
        {
            CompanyList = new List<CompanyModel>();
        }
    }

    public class CompanyEditModel
    {
        public int IdCompany { get; set; }
        public string IdAspNetUsers { get; set; } = string.Empty;
        public string Descrizione { get; set; } = string.Empty;
        public Boolean MainUser { get; set; }
    }
    public class CompanyGetInModel
    {
        public int Id { get; set; } = 0;
    }
    public class CompanyGetOutModel : ModelResult
    {
        public CompanyEditModel CompanyEdit { get; set; } = new CompanyEditModel();
    }
    public class CompanyPutInModel : ModelResult
    {
        public CompanyEditModel CompanyEdit { get; set; } = new CompanyEditModel();
    }
    public class CompanyPutOutModel : ModelResult
    {
        public CompanyEditModel CompanyEdit { get; set; } = new CompanyEditModel();
    }

}
